src/app/clusters/commissioner-control-server/commissioner-control-server.cpp

```
@@ -0,0 +1,277 @@
+    const Clusters::CommissionerControl::Commands::CommissionNode::DecodableType & commandData)
+{
+    CHIP_ERROR err = CHIP_NO_ERROR;
+
+    ChipLogProgress(Zcl, "Received command to commission node");
+
+    auto sourceNodeId = getNodeId(commandObj);
+
+    // Check if the command is executed via a CASE session
+    if (sourceNodeId == kUndefinedNodeId)
+    {
+        ChipLogError(Zcl, "Command not executed via CASE session, failing with UNSUPPORTED_ACCESS");
+        commandObj->AddStatus(commandPath, Status::UnsupportedAccess);
+        return true;
+    }
+
+    auto requestId = commandData.requestId;
+
+    auto commissionNodeInfo = std::make_unique<Clusters::CommissionerControl::CommissionNodeInfo>();
+
+    Clusters::CommissionerControl::Delegate * delegate =
+        Clusters::CommissionerControl::CommissionerControlServer::Instance().GetDelegate();
+
+    VerifyOrExit(delegate != nullptr, err = CHIP_ERROR_INCORRECT_STATE);
+
+    // Set IP address and port in the CommissionNodeInfo struct
+    commissionNodeInfo->port = commandData.port;
+    err                      = commissionNodeInfo->SetIPAddress(commandData.ipAddress);
+    SuccessOrExit(err == CHIP_NO_ERROR);
+
+    // Validate the commission node command.
+    err = delegate->ValidateCommissionNodeCommand(sourceNodeId, requestId);
+    SuccessOrExit(err == CHIP_NO_ERROR);
+
+    // Add the response for the commissioning window.
+    AddReverseOpenCommissioningWindowResponse(commandObj, commandPath, commissionNodeInfo->params);
+
+    // Schedule the deferred reverse commission node task
+    DeviceLayer::PlatformMgr().ScheduleWork(RunDeferredCommissionNode, reinterpret_cast<intptr_t>(commissionNodeInfo.release()));
+
+exit:
+    if (err != CHIP_NO_ERROR)
+    {
+        ChipLogError(Zcl, "emberAfCommissionerControlClusterCommissionNodeCallback error: %s", err.AsString());
+        commandObj->AddStatus(commandPath, Status::Failure);
```

Again, why are we suppressing the actual error?